src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/model/ManagedIndexMetaData.kt

```diff
@@ -109,7 +109,7 @@ data class ManagedIndexMetaData(
             builder.endObject()
         }
 
-        if (policyRetryInfo != null && !transitionToExists) {
+        if (policyRetryInfo != null) {
```

Comment:
Why are you removing this?

Reply:
When we set transitionTo, on the next execution we get the first action/firstStep of that next state. If that action is disallowed then we fail the policy (by just updating the metaData policy failed to true).

The metadata still has transitionTo set since we are not replacing it unless it actually transitioned. This means in the explain API the policy was failed but wasn't showing failed=true because a transitionTo existed.

This seems like it will happen for any policy failure on the first execution when transitioning, so we should probably always show PolicyRetryInfo regardless of transitionTo.

We still don't want to show action or state info because it refers to old state/action information (and if action/state information is updated it should mean the transitionTo was removed).